src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControl.java

```diff
@@ -11,75 +10,102 @@
                                         CaseService caseService,
                                         CaseTypeService caseTypeService,
                                         RoleAssignmentsFilteringService roleAssignmentsFilteringService,
-                                        FakeRoleAssignmentsGenerator fakeRoleAssignmentsGenerator,
+                                        PseudoRoleAssignmentsGenerator pseudoRoleAssignmentsGenerator,
                                         ApplicationParams applicationParams,
-                                        AccessProfileService accessProfileService) {
+                                        AccessProfileService accessProfileService,
+                                        PseudoRoleToAccessProfileGenerator pseudoRoleToAccessProfileGenerator) {
         this.roleAssignmentService = roleAssignmentService;
         this.securityUtils = securityUtils;
         this.caseService = caseService;
         this.caseTypeService = caseTypeService;
         this.roleAssignmentsFilteringService = roleAssignmentsFilteringService;
-        this.fakeRoleAssignmentsGenerator = fakeRoleAssignmentsGenerator;
+        this.pseudoRoleAssignmentsGenerator = pseudoRoleAssignmentsGenerator;
         this.applicationParams = applicationParams;
         this.accessProfileService = accessProfileService;
+        this.pseudoRoleToAccessProfileGenerator = pseudoRoleToAccessProfileGenerator;
     }
 
     // Returns Optional<CaseDetails>. If this is not enough think of wrapping it in a AccessControlResponse
     // that contains the additional information about the operation result
     @Override
     public Optional<CaseDetails> applyAccessControl(CaseDetails caseDetails) {
         RoleAssignments roleAssignments = roleAssignmentService.getRoleAssignments(securityUtils.getUserId());
-        CaseDetails cloned = caseService.clone(caseDetails);
 
         RoleAssignmentFilteringResult filteringResults = roleAssignmentsFilteringService
             .filter(roleAssignments, caseDetails);
 
-        if (filteringResults.hasGrantTypeExcludedRole()) {
-            filteringResults = filteringResults.retainBasicAndStandardGrantTypeRolesOnly();
-        }
-
         if (applicationParams.getEnablePseudoRoleAssignmentsGeneration()) {
-            List<RoleAssignment> augmentedRoleAssignments = fakeRoleAssignmentsGenerator
-                .addFakeRoleAssignments(filteringResults);
+            List<RoleAssignment> pseudoRoleAssignments = pseudoRoleAssignmentsGenerator
+                .createPseudoRoleAssignments(filteringResults);
+            filteringResults = augment(filteringResults, pseudoRoleAssignments);
         }
 
         CaseTypeDefinition caseTypeDefinition = caseTypeService.getCaseType(caseDetails.getCaseTypeId());
 
+        if (applicationParams.getEnablePseudoAccessProfilesGeneration()) {
+            List<RoleToAccessProfileDefinition> generatedAccessProfiles =
```

Comment:
you are not generating access profiles, but the mappings:
```suggestion
            List<RoleToAccessProfileDefinition> pseudoAccessProfilesMappings =
```